bears/typescript/TSLintBear.py

```diff
@@ -26,16 +52,170 @@ class TSLintBear:
 
     @staticmethod
     def create_arguments(filename, file, config_file,
-                         tslint_config: path="", rules_dir: path=""):
+                         tslint_config: path="",
+                         rules_dir: path="",
+                         directive_selectors_naming_convention: str="camel",
+                         component_selectors_naming_convention: str="hyphen",
+                         check_directive_selectors_type: bool=True,
+                         directive_selectors_type: str="attribute",
+                         check_component_selectors_type: bool=True,
+                         component_selectors_type: str="element",
+                         force_directive_selectors_prefix: bool=True,
+                         directive_selectors_prefix: str="sg",
+                         force_component_selectors_prefix: bool=True,
+                         component_selectors_prefix: str="sg",
+                         use_input_property_decorators: bool=True,
+                         use_output_property_decorators: bool=True,
+                         use_host_property_decorators: bool=True,
+                         allow_attribute_parameter_decorators: bool=False,
+                         allow_input_rename: bool=False,
+                         allow_output_rename: bool=False,
+                         allow_forward_references: bool=False,
+                         use_life_cycle_interface: bool=True,
+                         use_pipe_transform_interface: bool=True,
+                         pipe_naming_convention: str="camel",
+                         pipe_prefix: str="sg",
+                         force_pipe_prefix: bool=True,
+                         component_class_suffix: bool=True,
+                         directive_class_suffix: bool=True,
+                         check_import_destructuring_spacing: bool=True):
         """
-        :param tslint_config: Path to configuration file.
-        :param rules_dir:     Rules directory
+        :param tslint_config:
+            Path to configuration file. This will override all the settings
+            provided here.
+        :param rules_dir:
+            An additional rules directory, for user-created rules.
```

Comment:
not sure we should offer this, I'm thinking we may want to strip down the number of options we provide. Well it all goes down to my cEP :P we f*** need this